### PR TITLE
Origin and suggest added to Annotation

### PR DESCRIPTION
**Pull request by Holado**
Added origin and suggest to Annotation to facilitate comparison in eval.py. The values seem to be correctly added in errtokenizer.py but disappear somewhere along the way.

### CODE REVIEW

File: eval/eval.py [hunk 1]
```diff
@@ -1914,9 +1963,7 @@ def token_results(
                                 wrong_span += 1
                                 errtypefreqs[xtype]["wrong_span"] += 1
                             # 3. Error correction
-                            # Get the 'corrected' attribute if available,
-                            # otherwise use ytok['suggest']
-                            ycorr = getattr(ytok, "corrected", ytok.suggest)
+                            ycorr = getattr(ytok, "suggest", ytok.suggest)
```
Comment (vthorsteinsson) [inline eval/eval.py:1966]: The default here doesn't work - if `ytok` doesn't have the attribute `"suggest"`, it still tries to read `ytok.suggest`, which will fail.

File: eval/eval.py [hunk 2]
```diff
@@ -1963,7 +2010,7 @@ def token_results(
                     analysisblob.append("\tDumping rest of GC errors:")
                 while ytok is not None:
                     # This is a remaining GC annotation: false positive
-                    if ytok.code in ["E001"]:
+                    if ytok.code in GCSKIPCODES or "/w" in ytok.code:
```
Comment (vthorsteinsson) [inline eval/eval.py:2013]: Better: `ytok.code.endswith("/w")`

File: src/reynir_correct/annotation.py [hunk 3]
```diff
@@ -125,3 +128,15 @@ def suggest(self) -> Optional[str]:
...
+    @property
+    def original(self) -> Optional[str]:
+        """ The original text for the error """
+        return self._original
+
+    @property
+    def correction(self) -> Optional[str]:
+        """ A suggested correction for the token span, as a text string
+            containing tokens delimited by spaces """
+        return self._correction
```
Comment (vthorsteinsson) [inline src/reynir_correct/annotation.py:142]: Where is this property coming from? Doesn't seem to be assigned anywhere.

File: src/reynir_correct/checker.py [hunk 4]
```diff
@@ -275,6 +275,8 @@ def annotate(sent: _Sentence) -> List[Annotation]:
                             code=t.error_code,
                             text=t.error_description,
                             detail=t.error_detail,
+                            original=t.error_original if hasattr(t, "original") else "",
```
Comment (vthorsteinsson) [inline src/reynir_correct/checker.py:278]: Bug here: Should probably be `t.getattr("original", None)`  (or `t.getattr("error_original", None)`?)

File: src/reynir_correct/checker.py [hunk 5]
```diff
@@ -275,6 +275,8 @@ def annotate(sent: _Sentence) -> List[Annotation]:
...
+                            suggest=t.error_suggest if hasattr(t, "suggest") else ""
```
Comment (vthorsteinsson) [inline src/reynir_correct/checker.py:279]: Same comment as above, mutatis mutandis

File: src/reynir_correct/checker.py [hunk 6]
```diff
@@ -291,7 +293,9 @@ def annotate(sent: _Sentence) -> List[Annotation]:
                     code="E004",
                     text="Málsgreinin er sennilega ekki á íslensku",
                     detail="{0:.0f}% orða í henni finnast ekki í íslenskri orðabók"
-                        .format(words_not_in_bin/num_words * 100.0)
+                        .format(words_not_in_bin/num_words * 100.0),
+                    original="",
```
Comment (vthorsteinsson) [inline src/reynir_correct/checker.py:297]: Are these parameters required (keeping in mind that they default to `None`)?

File: src/reynir_correct/checker.py [hunk 7]
```diff
@@ -312,7 +316,9 @@ def annotate(sent: _Sentence) -> List[Annotation]:
                     code="E001",
                     text="Málsgreinin fellur ekki að reglum",
                     detail="Þáttun brást í kring um {0}. tóka ('{1}')"
-                        .format(err_index + 1, toktext)
+                        .format(err_index + 1, toktext),
+                    original="",
```
Comment (vthorsteinsson) [inline src/reynir_correct/checker.py:320]: Same comment as above

File: src/reynir_correct/errfinder.py [hunk 8]
```diff
@@ -209,6 +209,8 @@ def AðvörunHeldur(self, txt: str, variants: str, node: Node) -> AnnotationDict
         return dict(
             text="'{0}' er sennilega ofaukið".format(txt),
             detail="Yfirleitt nægir að nota 'en' í þessu samhengi.",
+            original=txt,
+            suggest=""
```
Comment (vthorsteinsson) [inline src/reynir_correct/errfinder.py:213]: `suggest=""` not required

File: src/reynir_correct/errfinder.py [hunk 9]
```diff
@@ -418,11 +433,12 @@ def AðvörunKomma(self, txt: str, variants: str, node: Node) -> AnnotationDict:
...
                 "('áður en ég fer [,] má sækja tölvuna') o.s.frv."
             ),
-            suggestion="",
+            original=",",
+            suggest="",
         )
 
     def VillaNé(self, txt: str, variants: str, node: Node) -> AnnotationDict:
-        return dict(text="'né' gæti átt að vera 'eða'", suggestion="eða")
+        return dict(text="'né' gæti átt að vera 'eða'", original="", suggest="eða")
```
Comment (vthorsteinsson) [inline src/reynir_correct/errfinder.py:441]: `original="né"`

File: eval/eval.py (hunk 1 above)
Comment (Holado) [reply eval/eval.py:1966]: Fixed

File: eval/eval.py (hunk 2 above)
Comment (Holado) [reply eval/eval.py:2013]: Fixed

File: src/reynir_correct/annotation.py (hunk 3 above)
Comment (Holado) [reply src/reynir_correct/annotation.py:142]: Leftovers from experiments, merged it with self._suggest

File: src/reynir_correct/checker.py (hunk 4 above)
Comment (Holado) [reply src/reynir_correct/checker.py:278]: Fixed

File: src/reynir_correct/checker.py (hunk 5 above)
Comment (Holado) [reply src/reynir_correct/checker.py:279]: Fixed

File: src/reynir_correct/checker.py (hunk 6 above)
Comment (Holado) [reply src/reynir_correct/checker.py:297]: No, they're not needed, at first I wanted to see what exactly I was dealing with, and explicitly state the values where I could. I'll delete them where possible.

File: src/reynir_correct/errfinder.py (hunk 9 above)
Comment (Holado) [reply src/reynir_correct/errfinder.py:441]: Fixed

File: eval/eval.py [hunk 10]
```diff
@@ -1898,7 +1898,10 @@ def token_results(
                             yorig = None
                         xorig = set(xtok["original"].split())
                         if ytok.suggest:
-                            ysugg = set(ytok.suggest.split())
+                            if type(ytok.suggest) == list:
```
Comment (vthorsteinsson) [inline eval/eval.py:1901]: Better:

```
if isinstance(ytok.suggest, str):
    ysugg = set(ytok.suggest.split())
else:
    ysugg = set(ytok.suggest)

```

File: src/reynir_correct/checker.py [hunk 11]
```diff
@@ -275,8 +275,8 @@ def annotate(sent: _Sentence) -> List[Annotation]:
                             code=t.error_code,
                             text=t.error_description,
                             detail=t.error_detail,
-                            original=t.error_original if hasattr(t, "original") else "",
-                            suggest=t.error_suggest if hasattr(t, "suggest") else ""
+                            original=t.error_original if hasattr(t, "error_original") else "",  # TODO or original
```
Comment (vthorsteinsson) [inline src/reynir_correct/checker.py:278]: Better: `original=t.getattr("error_original", "")`

File: src/reynir_correct/checker.py [hunk 12]
```diff
@@ -275,8 +275,8 @@ def annotate(sent: _Sentence) -> List[Annotation]:
...
+                            suggest=t.error_suggest if hasattr(t, "error_suggest") else ""      # TODO or suggest
```
Comment (vthorsteinsson) [inline src/reynir_correct/checker.py:279]: Ditto

File: eval/eval.py [hunk 13]
```diff
@@ -1748,21 +1748,21 @@ def process(fpath_and_category: Tuple[str, str]) -> Dict[str, Any]:
                                 # error with idx=dep_id
                                 dependencies.append((dep_id, error))
                             else:
-                                if QUIET:
+                                if not QUIET:
```
Comment (vthorsteinsson) [inline eval/eval.py:1751]: This code was actually correct. Normally, eval.py prints its input and output via `bprint()`, and error messages are interspersed within the output. But if the QUIET option is set, the output is not printed; only the error messages are. In that case, we want to show the relevant file name before each error message. Thus the `if QUIET:` clauses.

**Comment by vthorsteinsson** [comment]
Also, have a look at the mypy errors:

```
src/reynir_correct/errtokenizer.py:914: error: Argument 4 to "CompoundError" has incompatible type "List[str]"; expected "str"
src/reynir_correct/errtokenizer.py:1258: error: "CompoundError" gets multiple values for keyword argument "span"
src/reynir_correct/errtokenizer.py:1263: error: Argument 5 to "CompoundError" has incompatible type "str"; expected "int"
```

File: eval/eval.py (hunk 10 above)
Comment (Holado) [reply eval/eval.py:1901]: Fixed

File: src/reynir_correct/checker.py (hunk 11 above)
Comment (Holado) [reply src/reynir_correct/checker.py:278]: Fixed

File: src/reynir_correct/checker.py (hunk 12 above)
Comment (Holado) [reply src/reynir_correct/checker.py:279]: Fixed

File: eval/eval.py (hunk 13 above)
Comment (Holado) [reply eval/eval.py:1751]: My bad! Didn't mean to push this, did this to help with repeated runs. Will fix!

**Pull request merged**